lib/block-supports/elements.php

```diff
@@ -117,14 +125,26 @@ function gutenberg_render_elements_support_styles( $pre_render, $block ) {
 		$link_color      = "var(--wp--preset--color--$link_color_name)";
 	}
 	$link_color_declaration = esc_html( safecss_filter_attr( "color: $link_color" ) );
-
 	$style = ".$class_name a{" . $link_color_declaration . ';}';
 
+	if ( strpos( $link_hover_color, 'var:preset|color|' ) !== false ) {
+		// Get the name from the string and add proper styles.
+		$index_to_splice = strrpos( $link_hover_color, '|' ) + 1;
+		$link_hover_color_name = substr( $link_hover_color, $index_to_splice );
+		$link_hover_color      = "var(--wp--preset--color--$link_hover_color_name)";
+	}
+	$link_hover_color_declaration = esc_html( safecss_filter_attr(
+		"color: $link_hover_color"
+	) );
+	$style = ".$class_name a:hover{" . $link_hover_color_declaration . ';}';
```

Comment:
After https://github.com/WordPress/gutenberg/pull/40987 this might need a rebase.

Then we could get the link hover styles like so:

```php
$link_hover_block_styles = isset( $element_block_styles['link:hover'] ) ? $element_block_styles['link:hover'] : null;
$hover_styles = gutenberg_style_engine_generate(
    $link_hover_block_styles,
        array(
            'selector' => ".$class_name a:hover",
             'css_vars' => true,
        )
);

// Concat link and hover styles generated by style engine ...

```

I'm not sure yet, but I think one day the style engine could be updated to handle an elements object, so that we can return styles in one call, e.g., 

```php
$element_styles = gutenberg_style_engine_generate(
    array( 'elements' => array( 'link', => array( ... ), 'link:hover' => array( ... )  ) ),
        array(
            'selector' => ".$class_name a:hover",
             'css_vars' => true,
        )
);
```

Happy to help out here when the time arises. Thanks!

Reply:
Rebased, hopefully still works 😁

Reply:
Works! 🙇 


![2022-06-07 09 21 19](https://user-images.githubusercontent.com/6458278/172265280-b108fdb4-f1de-4559-98be-80198fe0ba0b.gif)